src/plugins/paced-transcript/index.js

```diff
@@ -46,26 +50,57 @@ function pacedTranscript(config) {
 
   // Generate captions from the transcription data
   const parseTranscript = transcriptionData => {
-    const maxWords = options.maxWords;
     const captions = [];
 
     transcriptionData.forEach(segment => {
       const words = segment.words;
+      const maxWords = options.maxWords || words.length;
 
       for (let i = 0; i < words.length; i += maxWords) {
-        const startTime = words[i].start_time;
-        const endTime = words[Math.min(i + maxWords - 1, words.length - 1)].end_time;
-
-        const captionText = words
-          .slice(i, i + maxWords)
-          .map(word => word.word)
-          .join(' ');
-
-        captions.push({
-          startTime: startTime,
-          endTime: endTime,
-          text: captionText
-        });
+        if (options.wordHighlight) {
+          // Create a caption for every word, in which the current word is highlighted
+          const slicedWords = words.slice(i, Math.min(i + maxWords, words.length));
+          slicedWords.forEach((word, idx) => {
+            const captionText = words
+              .slice(i, i + maxWords)
+              .map(w => (w === word ? `<b>${w.word}</b>` : w.word))
+              .join(' ');
+
+            captions.push({
+              startTime: word.start_time,
+              endTime: word.end_time,
+              text: captionText
+            });
+
+            // if we haven't reached the end of the words array, and there's a gap between the current word end_time and the next word start_time, add a non-highlighted caption to fill the gap
+            if (words[idx + 1] && word.end_time < words[idx + 1].start_time) {
+              const captionText = words
+                .slice(i, i + maxWords)
+                .map(word => word.word)
+                .join(' ');
+
+              captions.push({
+                startTime: word.end_time,
+                endTime: words[idx + 1].start_time,
+                text: captionText
+              });
+            }
+          });
+        } else {
+          const startTime = words[i].start_time;
+          const endTime = words[Math.min(i + maxWords - 1, words.length - 1)].end_time;
+
+          const captionText = words
+            .slice(i, i + maxWords)
+            .map(word => word.word)
+            .join(' ');
+
+          captions.push({
+            startTime: startTime,
+            endTime: endTime,
+            text: captionText
+          });
+        }
```

Comment:
I understand this problem is inherently imperative and i do not have a specific suggestion here, but see if you can simplify this function. It seems to me that same parts look like they can be reused.

Reply:
I agree, and adding the `timeOffset` makes this even more noticeable. re-factored